src/test/java/io/github/azagniotov/metrics/reporter/cloudwatch/CloudWatchReporterTest.java

```diff
@@ -278,6 +278,21 @@ public void shouldReportExpectedSingleGlobalDimension() throws Exception {
         assertThat(dimensions).contains(new Dimension().withName("Region").withValue("us-west-2"));
     }
 
+    @Test
+    public void shouldReportExpectedGlobalAndCustomDimensions() throws Exception {
+        metricRegistry.counter(new DimensionedNameBuilder(ARBITRARY_COUNTER_NAME)
+            .addDimension(new Dimension().withName("key1").withValue("value1"))
+            .addDimension(new Dimension().withName("key2").withValue("value2"))
+            .build().encode()).inc();
```

Comment:
Can the `.encode()` be called as part of `.build()`? Is there a use case where we would build a `DimensionedName` without encoding it? Also, a requirement to invoke the `.encode()` explicitly may open a door to errors where a client code may forget to call the API.

Thoughts?

Reply:
you can always do: `String myMetricName = dimensioned.encode()` and then use that one over and over.

Anyhow, I have done some refactoring, and also made the API a bit easier to use... might also give a reason to keep this as it is :-)